lib/practice_exercises.rb

```diff
@@ -1,13 +1,29 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n), regex results return a new array
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  processed_string = string.scan(/\w/)
+
+  return false if !(processed_string.length >= 0)
+
+  (processed_string.length/2).times do |x|
+    return false if processed_string[0+x].downcase != processed_string[-(1+x)].downcase
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) because nested times loops
+# Space Complexity: O(1), no additional data structures used
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍